src/coreclr/vm/jitinterface.cpp

```diff
@@ -1697,16 +1742,50 @@ void CEEInfo::getThreadLocalStaticBlocksInfo (CORINFO_THREAD_STATIC_BLOCKS_INFO*
 
     JIT_TO_EE_TRANSITION_LEAF();
 
-    pInfo->tlsIndex.addr = (UINT8*)0;
+    size_t threadStaticBaseOffset = 0;
 
-    pInfo->offsetOfThreadLocalStoragePointer = 0;
-    pInfo->offsetOfThreadStaticBlocks = 0;
-    pInfo->offsetOfMaxThreadStaticBlocks = 0;
-    pInfo->offsetOfGCDataPointer = 0;
+#if defined(TARGET_WINDOWS)
+    pInfo->tlsIndex.addr = (void*)static_cast<uintptr_t>(_tls_index);
+    pInfo->tlsIndex.accessType = IAT_VALUE;
 
-    JIT_TO_EE_TRANSITION_LEAF();
+    pInfo->offsetOfThreadLocalStoragePointer = offsetof(_TEB, ThreadLocalStoragePointer);
+    threadStaticBaseOffset = ThreadLocalOffset(&t_ThreadStatics);
+
+#elif defined(TARGET_OSX)
+
+    pInfo->threadVarsSection = getThreadVarsSectionAddress();
+
+#elif defined(TARGET_AMD64)
+
+    // For Linux/x64, get the address of tls_get_addr system method and the base address
+    // of struct that we will pass to it.
+    pInfo->tlsGetAddrFtnPtr = reinterpret_cast<void*>(&__tls_get_addr);
+    pInfo->tlsIndexObject = getTlsIndexObjectAddress();
```

Comment:
The method names are a mix of lowercase `getAAA` and uppercase `GetBBB`. It would be nice to make it uniform.

Reply:
`getAAA` methods are the ones in C++ and `GetBBB` is the one defined in `.S` file. I was thinking of having `Asm_GetBBB` but just went with `GetBBB`. let me know if you want me to do `Asm_*` naming to distinguish them.

Reply:
We also call `ThreadLocalOffset` that is implemented in C++, but starts with upper case.

I would make all global static start with upper case, irrespective whether they are implemented in C++ or asm.

(The instance JIT/EE interface methods should stay as is. The convention on JIT/EE interface is to start with lower case.)

Reply:
That makes sense.